### ISSUE TITLE
the stat of some layer is the same when using monitor.py

### ISSUE THREAD

**Issue by tornadomeet** [dmlc/mxnet#782]
@piiswrong i used monitor for debug today, and find some strange things, and i just run the [monitor_weights.py](https://github.com/dmlc/mxnet/blob/master/example/python-howto/monitor_weights.py),the output is like this:
```c++
 4 INFO:root:Batch:       1 fc1_output                     0.035887
   5 INFO:root:Batch:       1 relu1_output                   0.035887
   6 INFO:root:Batch:       1 fc2_output                     0.000129
   7 INFO:root:Batch:       1 relu2_output                   0.000129
   8 INFO:root:Batch:       1 fc3_output                     0.100000
   9 INFO:root:Batch:       1 softmax_output                 0.100000
  10 INFO:root:Batch:       1 softmax_backward_data          0.300001
  11 INFO:root:Batch:       1 softmax_backward_label         inf
  12 INFO:root:Batch:       1 fc3_backward_data              0.004012
  13 INFO:root:Batch:       1 fc3_backward_weight            0.006630
  14 INFO:root:Batch:       1 fc3_backward_bias              3.376438
  15 INFO:root:Batch:       1 relu2_backward_data            0.004012
  16 INFO:root:Batch:       1 fc2_backward_data              0.000128
  17 INFO:root:Batch:       1 fc2_backward_weight            0.001913
  18 INFO:root:Batch:       1 fc2_backward_bias              0.050234
  19 INFO:root:Batch:       1 relu1_backward_data            0.000128
  20 INFO:root:Batch:       1 fc1_backward_data              0.014501
  21 INFO:root:Batch:       1 fc1_backward_weight            0.000588
  22 INFO:root:Batch:       1 fc1_backward_bias              0.001619
  23 INFO:root:Batch:       1 data                           0.333816
  24 INFO:root:Batch:       1 fc1_weight                     0.005770
  25 INFO:root:Batch:       1 fc1_bias                       0.000002
  26 INFO:root:Batch:       1 fc2_weight                     0.005800
  27 INFO:root:Batch:       1 fc2_bias                       0.000050
  28 INFO:root:Batch:       1 fc3_weight                     0.005847
  29 INFO:root:Batch:       1 fc3_bias                       0.003376
  30 INFO:root:Batch:       1 softmax_label                  5.031898
  31 INFO:root:Epoch[0] Batch [100]  Speed: 39383.69 samples/sec     Train-accuracy=0.012667
  32 INFO:root:Batch:     101 fc1_output                     0.240131
  33 INFO:root:Batch:     101 relu1_output                   0.240131
  34 INFO:root:Batch:     101 fc2_output                     0.010978
  35 INFO:root:Batch:     101 relu2_output                   0.010978
  36 INFO:root:Batch:     101 fc3_output                     0.102843
  37 INFO:root:Batch:     101 softmax_output                 0.102843
  38 INFO:root:Batch:     101 softmax_backward_data          0.293702
  39 INFO:root:Batch:     101 softmax_backward_label         inf
  40 INFO:root:Batch:     101 fc3_backward_data              0.028850
  41 INFO:root:Batch:     101 fc3_backward_weight            0.749433
  42 INFO:root:Batch:     101 fc3_backward_bias              4.372302
```
1. the same stas output value between relu and it's previous  layer, in the whole training.
2. the same stas output value between softmax_output and it's previous layer, both lie in whole training.
3. some times it output inf value.
so is this normal?

**Comment by winstywang** [comment]
For 1, 2, it is normal because relu and softmax_output is inplace, that means it will share the same memory. 
For 3, I am not sure, it seems to be caused by the label. Could you output the label in the monitor to see what happened?

**Comment by piiswrong** [comment]
1,2 are actually bugs. I just fixed them. Monitors are injected in between inplace ops so that we can get all intermediary results.

3 is because we we don't backprop to label. So ignoring it should be fine.

**Comment by tornadomeet** [comment]
thanks, i got it
